### PR TITLE
[CLI] Move hostname, mgmt interface/vrf config to hostcfgd

### PR DESCRIPTION
**Pull request by fastiuk**
This PR depends on https://github.com/Azure/sonic-utilities/pull/2173

#### Why I did it
To be able to configure the management interface and hostname standalone by changing database config at runtime.
From the CLI perspective fo view, the following behavior is the same. But now you have two ways of configuring it: CLI, directly through the database.

#### How I did it
Moved configuration part of the interface and hostname to "hostcfgd".

#### How to verify it
* Built an image
* Flash it to the switch
* Run CLI commands
```
# Set IP address: verify address is set on the iface
sudo config interface ip add eth0 10.210.25.127/22 10.210.24.1
ip address show eth0
# 2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
#     link/ether 98:03:9b:a2:be:80 brd ff:ff:ff:ff:ff:ff
#     inet 10.210.25.127/22 brd 10.210.27.255 scope global eth0
#        valid_lft forever preferred_lft forever
#     inet6 fe80::9a03:9bff:fea2:be80/64 scope link
#       valid_lft forever preferred_lft forever

# Remove IP address: verify you received address form DHCP
sudo config interface ip remove eth0 10.210.25.127/22
ip address show eth0
# 2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
#     link/ether 98:03:9b:a2:be:80 brd ff:ff:ff:ff:ff:ff
#     inet 10.210.25.127/22 brd 10.210.27.255 scope global eth0
#        valid_lft forever preferred_lft forever
#     inet6 fe80::9a03:9bff:fea2:be80/64 scope link
#       valid_lft forever preferred_lft forever

# Enable/disable mgmt VRF
ip address show mgmt
# Device "mgmt" does not exist.

sudo config vrf add mgmt
ip address show mgmt
# 72: mgmt: <NOARP,MASTER,UP,LOWER_UP> mtu 65575 qdisc noqueue state UP group default qlen 1000
#     link/ether fa:9b:ad:7b:1e:83 brd ff:ff:ff:ff:ff:ff

sudo config vrf del mgmt
ip address show mgmt
# Device "mgmt" does not exist.

# Setting the hostname
admin@r-anaconda-27:~$ sudo config hostname bla
# Login / Logout
admin@bla:~$
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog
* Moved management interface configuration to hostcfgd.
* Moved management VRF configuration to hostcfgd.
* Moved hostname configuration to hostcfgd.

#### Submodules PR's :

| Repo  | PR title | State |
| ----------------- | ----------------- | ----------------- |
| sonic-utilities | [[CLI] Move hostname, mgmt interface/vrf config to hostcfgd](https://github.com/Azure/sonic-utilities/pull/2173) | GitHub issue/pull request detail |

### CODE REVIEW

**Comment by fastiuk** [comment]
@qiluo-msft @dgsudharsan @lguohan @prsunny @liat-grozovik
Moved here from this PR: https://github.com/Azure/sonic-buildimage/pull/10868

File: scripts/hostcfgd [hunk 1]
```diff
@@ -1253,6 +1254,143 @@ class PamLimitsCfg(object):
...
+
+        # Restart hostname-config service when hostname was changed.
+        # Empty not allowed
+        if new_hostname and new_hostname != self.hostname:
+            syslog.syslog(syslog.LOG_INFO, 'DeviceMetaCfg: Set new hostname: {}'
+                                           .format(new_hostname))
+            self.hostname = new_hostname
+            try:
+                run_cmd('sudo service hostname-config restart', True, True)
+            except Exception as e:
```
Comment (qiluo-msft) [inline scripts/hostcfgd:1290]: >Exception [](http://example.com/codeflow?start=19&length=9)

Could you use more specific exception type? The general type should only be used at the most outside level.

File: scripts/hostcfgd [hunk 2]
```diff
@@ -1253,6 +1254,143 @@ class PamLimitsCfg(object):
...
+
+        # Restart management interface service when config was changed
+        if data != self.iface_config_data.get(key):
+            cfg = {key: data}
+            syslog.syslog(syslog.LOG_INFO, f'MgmtIfaceCfg: Set new interface '
+                                           f'config {cfg} for {iface}')
+            try:
+                run_cmd('sudo systemctl restart interfaces-config', True, True)
+                run_cmd('sudo systemctl restart ntp-config', True, True)
+            except Exception:
```
Comment (qiluo-msft) [inline scripts/hostcfgd:1336]: >Exception [](http://example.com/codeflow?start=19&length=9)

Exception

File: scripts/hostcfgd [hunk 3]
```diff
@@ -1253,6 +1254,143 @@ class PamLimitsCfg(object):
...
+            return
+
+        syslog.syslog(syslog.LOG_INFO, f'Set mgmt vrf state {enabled}')
+
+        # Restart related vrfs services
+        try:
+            run_cmd('service ntp stop', True, True)
+            run_cmd('systemctl restart interfaces-config', True, True)
+            run_cmd('service ntp start', True, True)
+        except Exception:
```
Comment (qiluo-msft) [inline scripts/hostcfgd:1361]: >Exception [](http://example.com/codeflow?start=15&length=9)

The same.

**Review by qiluo-msft — APPROVED** [review]

**Comment by fastiuk** [comment]
This should be merged and should leave together with: https://github.com/sonic-net/sonic-utilities/pull/2173
I have no idea how they are living separately so far.

**Comment by liat-grozovik** [comment]
Once each will be merged you will need to have single pr for each of the submodules

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by qiluo-msft** [comment]
To clarify, if you want to move two submodules together, you can submit one PR to include both.

**Pull request merged**